Fixed default prop-types

9renpoto opened the pull request:
Fixed default prop types ( for TypeScript )

- [x] Added Types tests
- [x] Added Snapshot tests
- [x] Upgrade babel
- [x] Upgrade @storybook monorepo ( with *-loader )
- [x] Upgrade eslint 
- [x] coverage report post

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 193.32kb -> 154.62kb (20.02%)

/timeline.png -- 107.96kb -> 70.49kb (34.71%)
/stories/sample.jpg -- 42.68kb -> 42.06kb (1.45%)
/docs/ef2dd493e0f7c46d86759d30247aae16.jpg -- 42.68kb -> 42.06kb (1.45%)

9renpoto committed:
Fixed default prop-types

9renpoto commented on stories/App.story.js:
```diff
@@ -1,7 +1,7 @@
-import {storiesOf} from '@storybook/react'
-import {withInfo} from '@storybook/addon-info'
-import {Timeline, TimelineEvent, TimelineBlip} from '../components/index'
+import React, { Component } from 'react'
```
only format

9renpoto commented on .travis.yml:
```diff
@@ -5,10 +5,11 @@ cache:
 before_script:
   - npm prune
 after_success:
+  - '[ "$CMD" == "test" ] && npm run coverage'
```
Please enable codecov if you like

9renpoto commented:
@rcdexta Hi, please review it :)

9renpoto commented on stories/App.story.js:
```diff
@@ -197,48 +207,53 @@ storiesOf('Timeline', module)
   )
   .add(
-    'Dynamic Prop Updates',
+    '[DontTest] Dynamic Prop Updates',
```
Skip snapshot testing, because refs https://stackoverflow.com/questions/23199909/using-tolocalestring-in-node-js/23200062#23200062

9renpoto committed:
Storyshoots tests activate

9renpoto committed:
Added ignore coverage files

9renpoto committed:
Added badge

dependabot-support committed:
build(deps-dev): Bump @storybook/addon-options from 3.4.11 to 4.0.11

Bumps [@storybook/addon-options](https://github.com/storybooks/storybook) from 3.4.11 to 4.0.11.
- [Release notes](https://github.com/storybooks/storybook/releases)
- [Changelog](https://github.com/storybooks/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybooks/storybook/compare/v3.4.11...v4.0.11)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump eslint-plugin-promise from 2.0.1 to 4.0.1

Bumps [eslint-plugin-promise](https://github.com/xjamundx/eslint-plugin-promise) from 2.0.1 to 4.0.1.
- [Release notes](https://github.com/xjamundx/eslint-plugin-promise/releases)
- [Changelog](https://github.com/xjamundx/eslint-plugin-promise/blob/master/CHANGELOG.md)
- [Commits](https://github.com/xjamundx/eslint-plugin-promise/compare/v2.0.1...v4.0.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump @babel/preset-env from 7.1.6 to 7.2.0

Bumps [@babel/preset-env](https://github.com/babel/babel) from 7.1.6 to 7.2.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/master/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/compare/v7.1.6...v7.2.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump commitizen from 2.10.1 to 3.0.5

Bumps [commitizen](https://github.com/commitizen/cz-cli) from 2.10.1 to 3.0.5.
- [Release notes](https://github.com/commitizen/cz-cli/releases)
- [Commits](https://github.com/commitizen/cz-cli/compare/v2.10.1...v3.0.5)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump autoprefixer from 6.7.7 to 9.4.0

Bumps [autoprefixer](https://github.com/postcss/autoprefixer) from 6.7.7 to 9.4.0.
- [Release notes](https://github.com/postcss/autoprefixer/releases)
- [Changelog](https://github.com/postcss/autoprefixer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/autoprefixer/compare/6.7.7...9.4.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump @babel/cli from 7.1.5 to 7.2.0

Bumps [@babel/cli](https://github.com/babel/babel) from 7.1.5 to 7.2.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/master/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/compare/v7.1.5...v7.2.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump eslint-plugin-standard from 2.3.1 to 3.1.0

Bumps [eslint-plugin-standard](https://github.com/standard/eslint-plugin-standard) from 2.3.1 to 3.1.0.
- [Release notes](https://github.com/standard/eslint-plugin-standard/releases)
- [Commits](https://github.com/standard/eslint-plugin-standard/compare/v2.3.1...v3.1.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump cz-conventional-changelog from 1.2.0 to 2.1.0

Bumps [cz-conventional-changelog](https://github.com/commitizen/cz-conventional-changelog) from 1.2.0 to 2.1.0.
- [Release notes](https://github.com/commitizen/cz-conventional-changelog/releases)
- [Commits](https://github.com/commitizen/cz-conventional-changelog/compare/v1.2.0...v2.1.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump @storybook/addon-info from 3.4.11 to 4.0.11

Bumps [@storybook/addon-info](https://github.com/storybooks/storybook) from 3.4.11 to 4.0.11.
- [Release notes](https://github.com/storybooks/storybook/releases)
- [Changelog](https://github.com/storybooks/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybooks/storybook/compare/v3.4.11...v4.0.11)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump @babel/core from 7.1.6 to 7.2.0

Bumps [@babel/core](https://github.com/babel/babel) from 7.1.6 to 7.2.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/master/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/compare/v7.1.6...v7.2.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
build(deps-dev): Bump eslint-config-standard-react from 4.3.0 to 5.0.0

Bumps [eslint-config-standard-react](https://github.com/feross/eslint-config-standard-react) from 4.3.0 to 5.0.0.
- [Release notes](https://github.com/feross/eslint-config-standard-react/releases)
- [Commits](https://github.com/feross/eslint-config-standard-react/compare/v4.3.0...v5.0.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

9renpoto committed:
fix(stories) replace deprecated apis

dependabot-support committed:
Build(deps-dev): Bump semantic-release from 6.3.6 to 15.12.4

Bumps [semantic-release](https://github.com/semantic-release/semantic-release) from 6.3.6 to 15.12.4.
- [Release notes](https://github.com/semantic-release/semantic-release/releases)
- [Commits](https://github.com/semantic-release/semantic-release/commits/v15.12.4)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Build(deps-dev): Bump @storybook/storybook-deployer from 2.5.2 to 2.8.1

Bumps [@storybook/storybook-deployer](https://github.com/storybooks/storybook-deployer) from 2.5.2 to 2.8.1.
- [Release notes](https://github.com/storybooks/storybook-deployer/releases)
- [Changelog](https://github.com/storybooks/storybook-deployer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/storybooks/storybook-deployer/compare/v2.5.2...v2.8.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

rcdexta commented:
@9renpoto Thanks for doing this. This PR is massive and addresses lot of hygiene issues with this library. Let me take sometime to review and merge this!

9renpoto committed:
refactor(components) upgrade eslint

rcdexta commented on typings/index-tests.tsx:
```diff
@@ -0,0 +1,25 @@
+      process and pls do share the details asap. Consider this a gentle reminder
+      if you are on track already!
+    </TimelineEvent>
+  </Timeline>
```
Can we add `<TimelineBlip>` to the second event so that it is tested too?

9renpoto replied:
done 👍

rcdexta commented on package.json:
```diff
@@ -9,14 +9,15 @@
-    "test:cover": "istanbul cover -x *.test.js _mocha -- -R spec --require tests/config/setup 'tests/**/*.test.js'",
+    "test:story": "jest",
+    "test": "jest && tsc -p typings",
+    "test:watch": "jest -w",
```
should be `jest --watch` i guess. `-w` is used to set workers

rcdexta commented on typings/index-tests.tsx:
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
```
should add `@types/react` and `@types/react-dom` as dev dependencies to keep typescript compiler happy

9renpoto replied:
This library does not need to depend directly on `react-dom`, so it is not included.

rcdexta commented on package.json:
```diff
@@ -9,14 +9,15 @@
-    "test:watch": "mocha --require tests/config/setup 'tests/**/*.test.js' --watch",
-    "test:cover": "istanbul cover -x *.test.js _mocha -- -R spec --require tests/config/setup 'tests/**/*.test.js'",
+    "test:story": "jest",
+    "test": "jest && tsc -p typings",
```
what does `tsc -p typings` do in our case, since `typings` is already included in the npm release

9renpoto replied:
`index-tests.tsx` is a test by writing a simple sample and checking whether it can compile or type.

It is a method recommended mainly by `definitlyTyped`, and this time it also adopts` tsconfig`.

refs: https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute

rcdexta reviewed (changes requested):
Again thanks for your contributions. Please find minor review comments added. Happy to merge soon!

dependabot-support committed:
Build(deps-dev): Bump @types/react from 16.7.11 to 16.7.13

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped) from 16.7.11 to 16.7.13.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Build(deps-dev): Bump autoprefixer from 9.4.0 to 9.4.2

Bumps [autoprefixer](https://github.com/postcss/autoprefixer) from 9.4.0 to 9.4.2.
- [Release notes](https://github.com/postcss/autoprefixer/releases)
- [Changelog](https://github.com/postcss/autoprefixer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/autoprefixer/compare/9.4.0...9.4.2)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Build(deps-dev): Bump @storybook/react from 4.0.11 to 4.0.12

Bumps [@storybook/react](https://github.com/storybooks/storybook) from 4.0.11 to 4.0.12.
- [Release notes](https://github.com/storybooks/storybook/releases)
- [Changelog](https://github.com/storybooks/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybooks/storybook/compare/v4.0.11...v4.0.12)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-support committed:
Build(deps-dev): Bump @storybook/addon-info from 4.0.11 to 4.0.12

Bumps [@storybook/addon-info](https://github.com/storybooks/storybook) from 4.0.11 to 4.0.12.
- [Release notes](https://github.com/storybooks/storybook/releases)
- [Changelog](https://github.com/storybooks/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybooks/storybook/compare/v4.0.11...v4.0.12)

Signed-off-by: dependabot[bot] <support@dependabot.com>

9renpoto committed:
fix(package) use --watch options

9renpoto committed:
fix(typings) Added TimelineBlip types tests

9renpoto commented on typings/tsconfig.json:
```diff
@@ -0,0 +1,26 @@
+  },
+  "files": [
+    "index.d.ts",
+    "index-tests.tsx"
```
compile tests with type definitions

9renpoto commented:
Thank you for review. I fixed some. 👍
